src/malli/core.cljc

```diff
@@ -872,8 +872,8 @@
             (let [unparsers (into {} (map (fn [[k _ c]] [k (-unparser c)])) (-children this))]
               (fn [x]
                 (if (miu/-tagged? x)
-                  (if-some [unparse (get unparsers (key x))]
-                    (unparse (val x))
+                  (if-some [unparse (get unparsers (first x))]
+                    (unparse (second x))
```

Comment:
`nth` might be more equivalent in performance here since I think `first` needs to go via `seq`.